### PR TITLE
Require a min # of Ninja stars from a configured Team

### PR DESCRIPTION
**Pull request by casche**
:construction: 

Working on it. More to follow..

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/reviewninja/review.ninja/pull/1008" target="_blank">ReviewNinja

**Comment by casche** [comment]
closes #1006

File: src/client/controller/settings.js [hunk 1]
```diff
@@ -97,4 +97,10 @@ module.controller('SettingsCtrl', ['$scope', '$stateParams', '$HUB', '$RPC', '$m
             });
         };
 
+        $scope.teams = $RPC.call('repo', 'getTeams', {
```
Comment (dfarr) [inline src/client/controller/settings.js]: You can change this to `$HUB.call('orgs', 'getTeams')` and then we don't even need the new function in the repos API :smiley: !fix

Comment (casche) [inline src/client/controller/settings.js]: !fixed

Beauty! thanks :bow:

**Comment by casche** [comment]
@dfarr @brntbeer Kinda what you guys were thinking?

screen shot 2015-11-25 at 2 40 09 pm

**Comment by brntbeer** [comment]
@casche oh man. YES!

**Comment by dfarr** [comment]
:+1: and of course @rwreynard, thoughts?

**Comment by stephanzlatarev** [comment]
I added a separate feature request issue - #1010 . It was closed as a duplicate of this one. However, it is asking for a different use case that will not be covered by #1008 . Since I cannot re-open #1010 can you, please, check my comment there again and confirm that #1008 will cover the requested use case. If you agree it is different, please, reopen it.

In short, I would like to be able to enforce four eyes principle code review within the same team. #1008 will give me approval by a team that is authorized to merge pull requests but will not prevent anyone from that team to merge a pull request without consulting with a second person. For example, one person from the authorized team can create a new pull request, give a ninja star and then merge the pull request.

This can be implemented with a simple setting as either :
- (preferred by me) "[checkbox] For reviewninja/foo four eyes principle is required to merge. I.e. ninja stars by author do not count."
- (less preferred by me) "[checkbox] For reviewninja/foo authors cannot give ninja stars"

**Comment by brntbeer** [comment]
> For example, one person from the authorized team can create a new pull request, give a ninja star and then merge the pull request.

I'm not a maintainer, and certainly @dfarr or @casche should chime in, but I feel like this PR ( #1008 ) would sufficiently cover the goal of #1010. With this PR, if you have a minimum of 2 ninja stars required from a team that you're also on, you still are required to need someone who ISN'T you to confirm the code looks good.

**Comment by casche** [comment]
Hey @brntbeer, @dfarr and I discussed and we simplified :smirk: this to just adding the team dropdown. If no team is selected the repo requires x stars from 'Anyone'. If it's required, then people can still star but only the stars from the selected team can produce a green status. Still meets the use case, yes? 

screen shot 2015-12-02 at 4 49 34 pm

screen shot 2015-12-02 at 4 45 02 pm

**Comment by brntbeer** [comment]
@casche :heart_eyes: yes.

**Comment by pmn** [comment]
:sparkles::heart_eyes_cat::sparkles: This is very exciting

**Comment by jbjonesjr** [comment]
I could see a future where having 2 (or more) teams review code be helpful (QA team, docs team, and development manager, etc). The current approach is perfect for managing team (or unspecified) input, but could it be as simple as an "Add a new row" type ui to require multiple teams?

This may be way out of scope of the original request, if so, we can turn this into a separate issue.

**Comment by brntbeer** [comment]
> I could see a future where having 2 (or more) teams review code be helpful (QA team, docs team, and development manager, etc). The current approach is perfect for managing team (or unspecified) input, but could it be as simple as an "Add a new row" type ui to require multiple teams?

This is definitely outside of the scope of this PR. The workaround would be to ask 'why does it need to be 2 different teams. if teams are arbitrary groupings in organization, just make a mixed team "

**Comment by dfarr** [comment]
Agreed. At least for the first iteration of this feature let's keep it simple and unobtrusive. I like the "require x ninja stars from y", where y can be "anybody" or a specific team.

**Comment by dfarr** [comment]
We also need to think about our wording for the GitHub status, and the RN ui. Idea: "n ninja stars needed from team m". We can consistently show that _if_ a review team has been defined.

**Comment by brntbeer** [comment]
> We also need to think about our wording for the GitHub status, and the RN ui. Idea: "n ninja stars needed from team m". We can consistently show that if a review team has been defined.

@dfarr what if we used [the active voice](https://en.wikipedia.org/wiki/Active_voice) and said something like `n people from team M need to add ninja stars`

**Comment by dfarr** [comment]
:+1: like the active voice

**Comment by dfarr** [comment]
GitHub status (valid for all states):
```
n (of m) ninja stars [from t][, p (of q) problems resolved]
```

```
"from t" iff review team specified
", p (of q) problems resolved" iff q > 0
```

**Pull request merged**